### ISSUE TITLE
How to process sequence of images via in/out pipes?

### ISSUE THREAD

**Issue by iperov** [kkroening/ffmpeg-python#171]
```
job = ( ffmpeg
            .input("pipe:0", format="image2pipe")            
            .output("pipe:1", format="image2pipe")            
            .run_async(pipe_stdin=True, pipe_stdout=True)
           )           
    
    for filename in Path_utils.get_image_paths (input_path, ['.'+ext]):
        try:
            with open(filename, 'rb') as f:
                b = f.read()
        except:
            print("Unable to open file "+filename)
            continue
            
        job.stdin.write( b )
```
but job.stdout.read() freeze

**Comment by kkroening** [comment]
Check out the explanation [here](https://github.com/kkroening/ffmpeg-python/issues/156#issuecomment-449553709).

tl;dr: deadlocks are tricky.

**Comment by iperov** [comment]
problem is ffmpeg doesnot process given frame immediatelly and it wait several frames. Then a bunch of processed frames uploaded to stdout without any separators.
